mne/forward/tests/test_make_forward.py

```diff
@@ -354,4 +360,107 @@ def test_forward_mixed_source_space():
+        times += list(dfit.times)
+        pos += list(dfit.pos)
+        amplitude += list(dfit.amplitude)
+        ori += list(dfit.ori)
+        gof += list(dfit.gof)
+
+    # Create a new Dipole object with the dipole fits
+    dip_fit = Dipole(np.array(times), np.array(pos), np.array(amplitude),
+                     np.array(ori), np.array(gof))
+
+    # check that true (test) dipoles and fits are "close"
+    # cf. mne/tests/test_dipole.py
+    diff = dip_test.pos - dip_fit.pos
+    corr = np.corrcoef(dip_test.pos.ravel(), dip_fit.pos.ravel())[0, 1]
+    dist = np.sqrt(np.mean(np.sum(diff * diff, axis=1)))
+    gc_dist = 180 / np.pi * np.mean(np.arccos(
+            np.sum(dip_test.ori * dip_fit.ori, axis=1)))
+    amp_err = np.sqrt(np.mean((dip_test.amplitude - dip_fit.amplitude) ** 2))
+
+    # Make sure each coordinate is within 2 mm from reference
+    # NB tolerance should be set relative to snr of simulated evoked!
+    assert_allclose(dip_fit.pos, dip_test.pos, rtol=0, atol=2e-3,
+                    err_msg='position mismatch')
+    assert_true(dist < 2e-3, 'dist: %s' % dist)  # within 2 mm
+    assert_true(corr > 1 - 2e-3, 'corr: %s' % corr)
+    assert_true(gc_dist < 5, 'gc_dist: %s' % gc_dist)  # less than 5 degrees
+    assert_true(amp_err < 2e-3, 'amp_err: %s' % amp_err)  # within 2 nAm
+
+    # Make sure rejection works with bem: one dipole at z=1m
+    # NB _make_forward.py:_prepare_for_forward will raise a RuntimeError
+    # if no points are left after min_dist exclusions, hence 2 dips here!
+    dip_outside = Dipole(times=np.array([0., 0.001]),
+                         pos=np.array([[0., 0., 1.0], [0., 0., 0.040]]),
+                         amplitude=np.array([100e-9, 100e-9]),
+                         ori=np.array([[1., 0., 0.], [1., 0., 0.]]), gof=1)
+    assert_raises(ValueError, make_forward_dipole, dip_outside, fname_bem,
+                  evoked.info, fname_trans)
+
+    # Now make an evenly sampled set of dipoles, some simultaneous,
+    # should return a VolSourceEstimate regardless
+    times = np.array([0., 0., 0., 0.001, 0.001, 0.002])
+    pos = np.random.rand(6, 3)*0.020 + np.array([0., 0., 0.040])[np.newaxis, :]
+    amplitude = np.random.rand(6,)*100e-9
+    ori = np.eye(6, 3) + np.eye(6, 3, -3)
+    gof = np.arange(len(times))/len(times)  # arbitrary
+
+    dip_even_samp = Dipole(times, pos, amplitude, ori, gof)
+
+    # restrict sensors to EEG using pre-picking
+    picks = pick_types(info, meg=False, eeg=True)
+    eeg_info = pick_info(evoked.info, picks)
+    stc, fwd = make_forward_dipole(dip_even_samp, fname_bem, eeg_info,
+                                   trans=fname_trans)
+
+    assert_true(isinstance, VolSourceEstimate)
+    assert_allclose(stc.times, np.arange(0., 0.003, 0.001))
+    assert_true(fwd['info']['ch_names'][0] == 'EEG 001')
+    assert_true(fwd['info']['ch_names'][-1] == 'EEG 060')
+    # More(/more clever) tests on fwd?
```

Comment:
of course :) point taken about the heavy lifting being done elsewhere, will figure out where to shave off some seconds. lowest hanging fruit is just to test one dipole fit instead of (the arbitrarily chosen) five.